katello-attach-subscription

```diff
@@ -837,7 +837,11 @@ def vdcupdate()
     end
     system_details[system['name']]=sys
     # check if the type requested match the host one
-    system_type = KatelloAttachSubscription::FactAnalyzer.system_type(sys)
+    if sys.has_key?('facts')
```

Comment:
this shouldn't be necessary as we already use `system_type_from_host` in `system_type`:

https://github.com/RedHatSatellite/katello-attach-subscription/blob/0e6f39a31ec729461ecde8b32483347896f241c8/lib/katello_attach_subscription/fact_analyzer.rb#L39

Reply:
there was a bug with systems that have no facts at all, though, fixed in https://github.com/RedHatSatellite/katello-attach-subscription/pull/50